### PR TITLE
feat(server, web): partner share persons/faces

### PR DESCRIPTION
**Pull request by ttyridal**
This PR will bring the partner's persons/faces into the sharedTo-user's explorer UI
ref discussions: #7038 #5089 #6339 #5457 and probably more

The /person endpoint gains a _withPartners_ flag and web UI is updated to set it.

~~Remains to include _withPartners_ flag when requesting timeBuckets when _personId_ is set, but thought it would be nice to get some review going :)~~
Opportunistically including all parter ids when building time-bucket with the _personId_ option set could potentially be an overkill? 

The shared persons will still behave a bit "strange" (ref #7038) - they are read only!

### CODE REVIEW

File: server/src/domain/person/person.service.ts [hunk 1]
```diff
@@ -78,11 +80,19 @@ export class PersonService {
 
   async getAll(auth: AuthDto, dto: PersonSearchDto): Promise<PeopleResponseDto> {
     const { machineLearning } = await this.configCore.getConfig();
-    const people = await this.repository.getAllForUser(auth.user.id, {
+    const userIds: string[] = [auth.user.id];
+    if (dto.withPartners) {
+      const partners = await this.partnerRepository.getAll(auth.user.id);
+      const partnersIds = partners
+        .filter((partner) => partner.sharedBy && partner.sharedWith && partner.sharedById != auth.user.id) //TODO should && partner.inTimeline be included, as in assets?
```
Comment (danieldietzler) [inline server/src/domain/person/person.service.ts:87]: I think I would personally argue yes, however that's certainly debatable...

**Review by danieldietzler — COMMENTED** [review]
The code actually looks really good for now!

File: server/src/domain/person/person.service.ts (hunk 1 above)
Comment (ttyridal) [reply server/src/domain/person/person.service.ts:87]: Thanks for the review! 

I'm totally fine with both! Leaving the decision to someone with more milage on the project :)

Comment (ttyridal) [reply server/src/domain/person/person.service.ts]: kind of landed on the query parameter being sufficient, if you don't mind :)  

I think that, from a user expectation view, you're communicating your intent ("Also give me my partners photos/faces") pretty explicit if you add the query parameter.  requiring some magic setting in the database too would be "surprising behaviour".

**Review by martabal — COMMENTED** [review]
Mmmh, I don't know if it's that easy. How do you handle duplicates ? I mean, if you have pics of your son and your partner too, will it appear twice in the people list ?

Can you re-assign a face to a person owned by your partner ?

**Comment by ttyridal** [comment]
> Mmmh, I don't know if it's that easy. How do you handle duplicates ? I mean, if you have pics of your son and your partner too, will it appear twice in the people list ?

I specifically don't.  As stated in the PR message:

> The shared persons will still behave a bit "strange" (ref https://github.com/immich-app/immich/discussions/7038) - they are read only!

Yes, they will appear twice and no, it's not possible to merge own and partner's persons. I made this pull request to get the basic functionality going (and that will cover a lot of the referenced feature requests I think). Meanwhile, I started that discussion thread in the hope to get some discussion going on where to go next. 

In my opinion, a partner share should be near equivalent to your own libraries.. Though some might very likely disagree.

**Comment by martabal** [comment]
Right, I think it should be clear in the WebUI what you can modify and what you can't.

> In my opinion, a partner share should be near equivalent to your own libraries.. Though some might very likely disagree.

I agree but that's a technical challenge as soon as you have multiple partners : Let's say you have users A, B and C. If users A-B and A-C are partners, B and C must not see each other's faces / people.
Overall, it's a security and performance challenge

**Comment by ttyridal** [comment]
You are right, that will be a challenge. Suggest moving that discussion to #7038 as it's not really related to the PR (as it stand at least)

**Comment by ttyridal** [comment]
> Right, I think it should be clear in the WebUI what you can modify and what you can't.

Totally agree by the way! A little icon in the corner or something, indicating that this person object is shared to you would be awesome. (think I saw several request for various variants of that on other areas too in the discussion section by the way).. Not sure if I'm up to that task though.

**Comment by Yetangitu** [comment]
Does this PR have a future or is this functionality planned to be implemented in some other way?

**Comment by alextran1502** [comment]
@Yetangitu I was debating to merge the PR but what is stopping me is that this might cause confusion because the facial data isn't technically shared and synced between shared accounts. I think it would be better for us to wait and implement it the proper way

**Comment by ttyridal** [comment]
@alextran1502 Unfortunately I missed that debate.  Could you give a short summary of the objections? What's meant by "isn't technically shared and synced between shared accounts" f.ex?  or thoughts on the "proper way"..

Really want the functionality, so I would probably be willing to invest some time on this :)

**Comment by xttlegendapi** [comment]
this is a really important missing feature. anyone a update on this?

**Comment by sergey-litvinov** [comment]
I'm not sure it would help, but wanted to give another use case here. i have tons of photos made in the past and just want to host them and allow partner to see them as well. So in that case it's not about use it to add new photos, but more one person that would add fresh photos of both me and partner, and then second person would be viewing them. I agree that seeing duplicated faces might be confusing, but for this specific use case - it won't matter that much.

**Comment by derkling** [comment]
> @Yetangitu I was debating to merge the PR but what is stopping me is that this might cause confusion because the facial data isn't technically shared and synced between shared accounts. I think it would be better for us to wait and implement it the proper way

Any update on this topi? Could you please share a view on how this feature should be better implemented if the current PR is not good enough as a starting point?

**Comment by popy2k14** [comment]
Just my thoughts:
If i share my library with another person, i have assumed that also the faces data is shared.
I think it's an point of view.

Maybe just an checkbox during sharing would be enough, to give the user who is sharing the library an choice.

**Comment by Mirio** [comment]
+1 for this feature

**Comment by Apbooks** [comment]
+1 for this
Want to use Immich for family to upload old photos and identify people in the photos. Would like to have a shared album that users can discuss and change the names and dates and locations. Would be really cool to be able to collaborate with all users and have comments about the photos. Sharing faces would make this really awesome!

**Comment by alextran1502** [comment]
As we are working toward separating the asset relationship, which will ultimately help to solve this use case, I will be closing this PR since it is stale and not the direction we would take to handle this sharing mechanism

**Comment by srvprivate** [comment]
@alextran1502 trying to follow this chain, I've been deeply involved in family photos from multiple sides now, just lost my grandpa recently and 2 others last year, that's the one thing I'm waiting on, so I can create EVERYONE a user account. It makes tracking uploads easier too. But unless I'm missing something, having separate accounts won't let faces be shared, so that's why I've been waiting. What's the progress? Percentage wise? What's your scrum look like? Thanks man, REALLY love all the hard work you do, my family really love this, especially in the wake of death, it was very touching for them. (broken family, trying to repair, this has been something that does that). God bless!

**Comment by danieldietzler** [comment]
@srvprivate we haven't started working on it yet, #12614 is the overarching issue for this.

**Comment by srvprivate** [comment]
> @srvprivate we haven't started working on it yet, #12614 is the overarching issue for this.

Thanks @danieldietzler makes sense after reading it. We're gridlocked! I made another comment on Sharing Libraries so yeah it seems we're all in agreement. #7362 was where I'd detailed my troubles. I've put a stop on inviting family until user accounts can all talk to what needs to be talked about. Thanks for pointing me in the right direction. Are there any other discussions I should be following concerning these talks?

**Comment by danieldietzler** [comment]
Not really. I think we have a pretty good idea of what people want at this point; we haven't really seen any novel ideas for that in a while. That issue I linked is definitely the one to follow as we will reference that in related PRs.

**Comment by srvprivate** [comment]
> Not really. I think we have a pretty good idea of what people want at this point; we haven't really seen any novel ideas for that in a while. That issue I linked is definitely the one to follow as we will reference that in related PRs.

I have misspoken. Thanks for making that clear, I will wait patient and just continue to watch! Thanks for all of your all's hard work, I'll get out of the way.

**Pull request closed without merging**